crypto/rsa/rsa_sp800_56b_gen.c

```diff
@@ -209,6 +209,131 @@ static int rsa_validate_rng_strength(EVP_RAND_CTX *rng, int nbits)
     return 1;
 }
 
+
+DEFINE_STACK_OF(BIGNUM)
+
+/*
+ *
+ * Using p & q, calculate dmp1, dmq1, iqmp
+ * Placing them on their respective BIGNUM stacks
+ *
+ * See SP800-56Br1
+ *   6.3.1.1 rsakpg1 - basic (Steps 3-4)
+ *   6.3.1.3 rsakpg1 - crt   (Step 5)
+ *
+ * Params:
+ *  p,q - input primes
+ *  d,n - input key values
+ *  e - input modulus
+ *  nbits - requested size of our computed values
+ *  exps, coeffs - output list of computed values for dmp1, dmq1, iqmp
+ *  modify - boolean value to indicate if the provided inputs should be modified in place
+ *           see use in ossl_rsa_sp800_56b_derive_params_from_pq
+ *
+ * Returns: -1 = error,
+ *           0 = d is too small,
+ *           1 = success.
+ *
+ */
+int ossl_rsa_sp800_56b_build_exp_coeffs_from_pq(BIGNUM *p, BIGNUM *q, BIGNUM *d, BIGNUM *n, int nbits,
+                                             const BIGNUM *e, STACK_OF(BIGNUM) *exps, STACK_OF(BIGNUM) *coeffs, BN_CTX *ctx, int modify)
+{
+    int ret = -1;
+    BIGNUM *p1, *q1, *lcm, *p1q1, *gcd;
+    BIGNUM *dmp1, *dmq1, *iqmp;
+    BIGNUM *dp, *dq, *dd, *dn;
+
+    BN_CTX_start(ctx);
+    p1 = BN_CTX_get(ctx);
+    q1 = BN_CTX_get(ctx);
+    lcm = BN_CTX_get(ctx);
+    p1q1 = BN_CTX_get(ctx);
+    gcd = BN_CTX_get(ctx);
+
+    if (modify == 0) {
+        dp = BN_dup(p);
+        dq = BN_dup(q);
+        dd = BN_dup(d);
+        dn = BN_dup(n);
+    } else {
+        dp = p;
+        dq = q;
+        dd = d;
+        dn = n;
+    }
+
+    if (!dp || !dq || !dd || !dn)
```

Comment:
nit: == NULL

Reply:
this is removed with my un-refactoring of the derivation function